packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts

```diff
@@ -75,43 +75,38 @@ export function useUploadFiles({
       };
 
       if (file) {
-        const {
-          file: processedFile,
-          key: processedKey,
-          ...processedRest
-        } = isFunction(processFile)
-          ? processFile({ file, key })
-          : { file, key };
-
-        onUploadStart?.({ key });
-
-        if (isResumable) {
-          const uploadTask = uploadFile({
-            ...processedRest,
-            file: processedFile,
-            fileName: path + processedKey,
-            isResumable: true,
-            level: accessLevel,
-            completeCallback: onComplete,
-            progressCallback: onProgress,
-            errorCallback: onError,
-            provider,
-          }) as unknown as UploadTask;
-          setUploadingFile({ id, uploadTask });
-        } else {
-          uploadFile({
-            ...processedRest,
-            file: processedFile,
-            fileName: path + processedKey,
-            isResumable: false,
-            level: accessLevel,
-            completeCallback: onComplete,
-            progressCallback: onProgress,
-            errorCallback: onError,
-            provider,
-          });
-          setUploadingFile({ id });
-        }
+        resolveFile({ processFile, file, key }).then(
+          ({ file, key, ...rest }) => {
+            onUploadStart?.({ key });
+            if (isResumable) {
+              const uploadTask = uploadFile({
+                ...rest,
+                file,
+                key: path + key,
+                isResumable: true,
+                level: accessLevel,
+                completeCallback: onComplete,
+                progressCallback: onProgress,
+                errorCallback: onError,
+                provider,
+              }) as unknown as UploadTask;
+              setUploadingFile({ id, uploadTask });
+            } else {
+              uploadFile({
+                ...rest,
+                file,
+                key: path + key,
+                isResumable: false,
+                level: accessLevel,
+                completeCallback: onComplete,
+                progressCallback: onProgress,
+                errorCallback: onError,
+                provider,
+              });
+              setUploadingFile({ id });
+            }
```

Comment:
Think this could be simplified to:
```suggestion
            const uploadTask = uploadFile({
              ...rest,
              file,
              key: path + key,
              isResumable,
              level: accessLevel,
              completeCallback: onComplete,
              progressCallback: onProgress,
              errorCallback: onError,
              provider,
            }) as unknown as UploadTask;
            setUploadingFile({ id, uploadTask: isResumable ? uploadTask : undefined });
```